CountersAndLists in Reference Documentation (#3529)
googleforgames/agones: site/content/en/docs/Reference/fleet.md

      version: "" # empty an existing label value
    annotations:
      otherkey: setthisvalue
  #

>   # [Stage:Alpha]

Reviewer: Nice!

Please also add a section below "The `spec` field is the actual `Fleet` specification and it is composed as follow:" which is where we also explain part of the new data structure. it's basically a repeat of above in a slightly different format.

Also same with each of the other `Reference/*.md` files in this PR.

Reviewer: Sorry - that wasn't what I meant.

I'm talking about this section here: 

![image](https://github.com/googleforgames/agones/assets/298370/6e3f9267-3181-40c7-84eb-da3ee4af492c)

This is where we also explain what each part of the resource (Fleet, GameServer, etc) configuration does, and extra details about that.

Does that make more sense?

Reviewer: So to be extra clear - revert this commit: https://github.com/googleforgames/agones/pull/3529/commits/59900ca5718d9420d48c0402f0240a7dab5e77ac - you had it right in the first place 👍🏻

Reviewer 2: +1 to this, add a note on `priorities` before `template` https://github.com/googleforgames/agones/blob/735e1cf069ffcdf10ff37b2c922e4215da5cf70d/site/content/en/docs/Reference/fleet.md?plain=1#L120 and after `allocationOverflow` to keep with the same ordering as the example yaml.

Reviewer: @Kalaiselvi84 just a reminder that this also needs the section at the bottom of the docs here:
https://agones.dev/site/docs/reference/gameserver/#:~:text=The%20spec%20field%20is%20the%20actual%20GameServer%20specification%20and%20it%20is%20composed%20as%20follow%3A

      version: "" # empty an existing label value
    annotations:
      otherkey: setthisvalue
  #
  # [Stage:Alpha]
  # [FeatureFlag:CountsAndLists]
  # Which gameservers in the Fleet are most important to keep around - impacts scale down logic.

>   # counters:

Reviewer 2: ```suggestion
  # priorities:
```

>   #   - type: Counter # Sort by a “Counter”

Reviewer 2: ```suggestion
  # - type: Counter # Sort by a “Counter”
```

>   #     key: player # The name of the Counter. No impact if no GameServer found.

Reviewer 2: ```suggestion
  #   key: player # The name of the Counter. No impact if no GameServer found.
```

>   #     order: Descending # Default is "Ascending" so smaller capacity will be removed first on down scaling.

Reviewer 2: ```suggestion
  #   order: Descending # Default is "Ascending" so smaller capacity will be removed first on down scaling.
```

>   # lists:

Reviewer 2: ```suggestion
```

>   #   - type: List # Sort by a “List”

Reviewer 2: ```suggestion
  # - type: List # Sort by a “List”
```

>   #     key: room # The name of the List. No impact if no GameServer found.

Reviewer 2: ```suggestion
  #   key: room # The name of the List. No impact if no GameServer found.
```

>   #     order: Ascending # Default is "Ascending" so smaller capacity will be removed first on down scaling.

Reviewer 2: ```suggestion
  #   order: Ascending # Default is "Ascending" so smaller capacity will be removed first on down scaling.
```

  - `Fleet's Scheduling Strategy`: The GameServers associated with the GameServerSet are sorted based on either `Packed` or `Distributed` strategy.
      - `Packed`: Agones maximizes resource utilization by trying to populate nodes that are already in use before allocating GameServers to other nodes.
      - `Distributed`: Agones employs this strategy to spread out GameServer allocations, ensuring an even distribution of GameServers across the available nodes.
- `priorities`: (Alpha, requires `CountsAndLists` feature flag): Defines which gameservers in the Fleet are most important to keep around - impacts scale down logic.
  - `type`: Sort by a "Counter" or a "List".

>     - `key`: The name of the Counter or List. If not found on the GameServer, has no impact.

Reviewer 2: ```suggestion
  - `key`: The name of the Counter or List. If not found on the GameServer, has no impact.
```

>     - `order`: Order: Sort by “Ascending” or “Descending”. “Descending” a bigger Capacity is preferred. “Ascending” would be smaller Capacity is preferred.

Reviewer 2: ```suggestion
  - `order`: Order: Sort by “Ascending” or “Descending”. “Descending” a bigger Capacity is preferred. “Ascending” would be smaller Capacity is preferred.
```

  - `Fleet's Scheduling Strategy`: The GameServers associated with the GameServerSet are sorted based on either `Packed` or `Distributed` strategy.
      - `Packed`: Agones maximizes resource utilization by trying to populate nodes that are already in use before allocating GameServers to other nodes.
      - `Distributed`: Agones employs this strategy to spread out GameServer allocations, ensuring an even distribution of GameServers across the available nodes.

> - `priorities` (Alpha, requires `CountsAndLists` feature flag): Defines which gameservers in the Fleet are most important to keep around - impacts scale down logic.

Reviewer: Tweaking some formatting and docs, but this is exactly what we need here and in the other .md files in this PR 👍🏻 

```suggestion
- `priorities`: (Alpha, requires `CountsAndLists` feature flag): Defines which gameservers in the Fleet are most important to keep around - impacts scale down logic.
```

>   - `type: Counter`: Sorts GameServers by a "counter".
>     - `key`: The name of the Counter. No impact if no GameServer found.
>     - `order`: The sort order, default is "Ascending" so smaller capacity will be removed first on down scaling.

Reviewer: Lifting from http://localhost:1313/docs/reference/agones_crd_api_reference/#agones.dev/v1.Priority

```suggestion
  - `type`: Sort by a “Counter” or a “List”.
    - `key`: The name of the Counter or List. If not found on the GameServer, has no impact.
    - `order`: Order: Sort by “Ascending” or “Descending”. “Descending” a bigger Capacity is preferred. “Ascending” would be smaller Capacity is preferred.
```

>   - `type: List`: Sorts GameServers based on a list metric.
>     - `key`: The name of the List. No impact if no GameServer found.
>     - `order`: The sort order, default is "Ascending" so smaller capacity will be removed first on downscaling.

Reviewer: ```suggestion
```
